ndscan/experiment/default_analysis.py

```diff
@@ -245,103 +296,130 @@ def execute(
+        some point in the future we could consider choosing sensible default scale
+        factors, but it's not totally trivial to do that in a way that doesn't blow up
+        for parameters with initial values close to zero.
     """
     def __init__(self,
-                 fit_type: str,
+                 fit_class: str,
                  data: Dict[str, Union[ParamHandle, ResultChannel]],
                  annotations: Optional[Dict[str, Dict[str, Any]]] = None,
                  analysis_identifier: str = None,
                  constants: Optional[Dict[str, Any]] = None,
-                 initial_values: Optional[Dict[str, Any]] = None):
-        self.fit_type = fit_type
-        if fit_type not in FIT_OBJECTS:
-            logger.warning("Unknown fit type: '%s'", fit_type, exc_info=True)
+                 initial_values: Optional[Dict[str, Any]] = None,
+                 bounds: Optional[Dict[str, Tuple[float, float]]] = None,
+                 exported_results: Optional[List[ExportedResult]] = None,
+                 fit_module: str = 'oitg.fitting',
+                 scale_factors: Optional[Dict[str, float]] = None):
+
+        self.fit_class_name = fit_class
+        self.fit_module = fit_module
         self.data = data
-        if annotations is None:
-            annotations = DEFAULT_FIT_ANNOTATIONS.get(fit_type, {})
         self.annotations = annotations
         self.analysis_identifier = analysis_identifier
-        self.constants = {} if constants is None else constants
-        self.initial_values = {} if initial_values is None else initial_values
+        self.initial_values = initial_values or {}
+        self.exported_results = exported_results or []
+        self.scale_factors = scale_factors or {}
+
+        self._result_channels = []
+        for result in self.exported_results:
+            channel = result.result_type(path=result.result_name, **result.config)
+            self._result_channels.append(channel)
+
+            if result.export_err:
+                err_channel = FloatChannel(
+                    path=result.result_name + '_err',
+                    min=0.0,
+                    display_hints={"error_bar_for": channel.path})
+                self._result_channels.append(err_channel)
+
+        duplicate_channels = [
+            channel.path
+            for channel, count in collections.Counter(self._result_channels).items()
+            if count > 1
+        ]
+        if duplicate_channels:
+            raise ValueError(
+                f"Duplicate result channels: {','.join(duplicate_channels)}")
+        self._result_channels = {
+            channel.path: channel
+            for channel in self._result_channels
+        }
+
+        klass = utils.import_class(self.fit_module, self.fit_class_name)
+        self.fit_klass = klass
+        self.bounds = bounds if bounds is not None else klass.get_default_bounds()
```

Comment:
hmmm...should the bounds replace the defaults completely or just overwrite specific ones 🤔 . Same in the fitting code. Same question for constants etc...